Initial Graphics Test

jgross11 opened the pull request:
hello

jgross11 committed:
Update routes.py

graphics test routing

jgross11 committed:
Create graphicstest.jinja2

initial p5 work to draw simple board

jgross11 committed:
Update graphicstest.jinja2

image loading and rendering

jgross11 committed:
Update graphicstest.jinja2

piece moves to / when mouse is pressed

robbycw reviewed (commented):
Looks like a good start! It's nice being able to move the king around. Were we still thinking of eventually having configurable sizes to the board?

jgross11 commented:
> Looks like a good start! It's nice being able to move the king around. Were we still thinking of eventually having configurable sizes to the board?

Yep - this is mainly just to show basic functionality with P5. We'll need a very fancy system in place to handle drawing variably sized boards, detecting bounds for clicking + dragging pieces, etc. The game model will have all this info, and the view will (probably) handle the rendering of that model while also performing visual updates to it while waiting for the backend to confirm a change (ex. moving a piece to a different square)..

jgross11 committed:
Create Game.js

template for what a game object received from backend might look like

jgross11 committed:
Create ViewUtils.js

:^)

jgross11 committed:
Update graphicstest.jinja2

somewhat flexible view and logic handler for board given assumptions about game object

jgross11 committed:
comments to pad my contribution count

jgross11 commented:
start of #41

andersonjackc commented on bless_this_chess/view/static/js/ViewUtils.js:
```diff
@@ -0,0 +1,2 @@
+// thanks js
+intdiv = (a,b) => (a-a%b)/b;
```
👀

jgross11 replied:
👀

andersonjackc commented:
This is really starting to look great.  It was fun being able to capture pieces and have jumps that you wouldn't get in a normal game of chess, which is what we will eventually want anyway!

Thanks for getting work started on this again!

andersonjackc commented:
Found a bug where if you pick up your piece and then just let go it deletes itself.

jgross11 committed:
Update graphicstest.jinja2

fix off board positioning, self destroying bug

jgross11 commented:
> Found a bug where if you pick up your piece and then just let go it deletes itself.

good catch buddy
